src/Components/Web.JS/src/Services/NavigationEnhancement.ts

```diff
@@ -48,6 +48,8 @@ export function attachProgressivelyEnhancedNavigationListener(callbacks: Navigat
   document.addEventListener('click', onDocumentClick);
   document.addEventListener('submit', onDocumentSubmit);
   window.addEventListener('popstate', onPopState);
+
+  attachProgrammaticEnhancedNavigationHandler(performProgrammaticEnhancedNavigation);
```

Comment:
I know that previously you were also thinking of adding a "removeProgressivelyEnhancedNav" capability in this PR that would deal with cases where someone dynamically removes the `Router` (e.g., doing an enhanced nav to a page that doesn't contain it).

Is that still something you want to address in RC1/RC2 or do you now think we're OK without it? I'm very open to different outcomes here, and just want to know what your current thinking is.

Reply:
I've opened https://github.com/dotnet/aspnetcore/issues/50069 to track a similar problem, which is that navigation interception doesn't get disabled when the router gets removed. This causes other problems, but shouldn't be an issue for this feature in particular.

I prototyped an implementation in https://github.com/dotnet/aspnetcore/pull/50012, but that PR got closed in favor of this one. I want to keep the changes in this PR minimal given that it's (hopefully) going into RC1 via ask mode, but I think that issue should be addressed in RC2.